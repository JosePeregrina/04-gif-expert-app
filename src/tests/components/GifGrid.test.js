import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import GifGrid from '../../Components/GifGrid';
import { useFetchGifts } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('El componente <GifGrid />', () => {
    const category = 'One Punch Man'
  test('debe cargar correctamente ', () => {

    useFetchGifts.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('debe mostar items cuando se cargar imagenes con usefetchGifs', () => {
    const gifs = [{
        id: 'abc',
        title: 'cualquier cosa',
        url: 'http://base_url.com',
      },{
        id: 'def',
        title: 'cualquier otra cosa',
        url: 'http://base_url.com',
      }]
    useFetchGifts.mockReturnValue({
        data: gifs,
        loading: false,
      });
    const wrapper = shallow(<GifGrid category={category} />);

    expect( wrapper.find('p').exists() ).toBe(false);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    
  });
});
