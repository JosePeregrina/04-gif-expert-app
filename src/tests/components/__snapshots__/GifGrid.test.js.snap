// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`El componente <GifGrid /> debe cargar correctamente  1`] = `
<Fragment>
  <h3>
    One Punch Man
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`El componente <GifGrid /> debe mostar items cuando se cargar imagenes con usefetchGifs 1`] = `
<Fragment>
  <h3>
    One Punch Man
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquier cosa"
      url="http://base_url.com"
    />
    <GifGridItem
      id="def"
      key="def"
      title="cualquier otra cosa"
      url="http://base_url.com"
    />
  </div>
</Fragment>
`;
